### ISSUE TITLE
Pokemon Super Mystery Dungeon: Bad RGBA8 texture pointer in H/W renderer

### ISSUE THREAD

**Issue by j-selby** [citra-emu/citra#2110]
At commit: 7ed97fb
Game save (just beyond the personality test): [00174600.zip](https://github.com/citra-emu/citra/files/508427/00174600.zip)

_This is unrelated to the CAM service timing issue. The camera is only used on the first launch and can be avoided by using a savegame._

Super Mystery Dungeon crashes when attempting to read a bad texture pointer. This occurs every time the opening sequence is viewed, about ~30 seconds in. This only occurs in the H/W renderer. This also occurs in exactly the same manner in at least the next scene, involving your character's reflection in the water.

Log - nothing before this point, and immediate crash after outputting this (on further analysis, this also appears irrelevant, as other occurrences don't throw any warnings):

```
[  63.809003] HW.GPU <Critical> core\hw\gpu.cpp:GPU::TextureCopy:336: zero input width
[  63.809657] HW.GPU <Critical> core\hw\gpu.cpp:GPU::TextureCopy:336: zero input width
```

Screenshots:

![devenv-04-10-2016_23-33-40](https://cloud.githubusercontent.com/assets/1404334/19074284/38f62eea-8a8b-11e6-9946-f45c586e9dc7.png)
![devenv-04-10-2016_23-34-07](https://cloud.githubusercontent.com/assets/1404334/19074285/38f80260-8a8b-11e6-8912-f1ae67ec9529.png)
![nvidia share-04-10-2016_23-25-58](https://cloud.githubusercontent.com/assets/1404334/19074286/38faa704-8a8b-11e6-9ca1-8b416c57285c.png)

**Comment by majormatheus** [comment]
Already have some solution to this problem?

**Comment by cjdc88** [comment]
Experience crash at the end of talking to gabite in drilbur mine (unable to fight him) and always crashes at the same spot throwing this as the last 2 lines.

 core\hw\gpu.cpp:TextureCopy:343: zero input width
core\hw\gpu.cpp:TextureCopy:343: zero input width  

any fix yet? :v

**Comment by techwolfe** [comment]
Hopefully there is a fix for this cause I really want to play the game, and my 3ds is broken...

**Comment by Hexagon12** [comment]
For now, what I do to stop these crashes, is disabling H/W renderer, until it doesn't throw anymore "zero input width" messages. Quick tips, where these crashes occur and when you should disable the H/W:

1. Opening sequence, when Rayquaza flies into Deoxys.
2. When you look into the water reflection (the whole scene)
3. In every boss battle (the whole scene with white triangles from sides)
4. When you are attacked by a Poliwrath after the battle.
5. At night, when Watchog is on guard duty, and somebody says "ghosts".

**Comment by Hexagon12** [comment]
This also error also happens on Gates To Infinity, but it doesn't crash. Same error message, but it doesn't crash.

**Comment by knife-edge** [comment]
Pokemon Super Mystery Dungeon Crashes:

All as mentioned by @Hexagon12 and the following:
After the first watchdog scene, when Audino tells Farfetch'd about ghosts.

**Comment by Helios747** [comment]
test

**Comment by JayFoxRox** [comment]
[The existing posts which help users to workaround this issue have been moved into a new forum topic](https://community.citra-emu.org/t/pokemon-super-mystery-dungeon-crash-locations/1465).
(If a developer is looking into replicate this issue I'd also recommend checking that topic.)

**Comment by IchigoMait** [comment]
The game plays now fine without crashing while using hardware renderer, I'm using the Citra Bleeding Edge build 0.1.405 (HEAD-c017065). Thx for fixing this wwylele.

**Comment by hcorion** [comment]
Was apparently yuriks that fixed this, via https://github.com/citra-emu/citra/pull/2799

**Comment by j-selby** [comment]
Fix hasn't been merged yet into master, hold your horses.

**Comment by Gynlod23** [comment]
what do i do for this problem? i dont understand

**Comment by IchigoMait** [comment]
@Gynlod23 Download Citra that is version 2799 at least, I'm using Citra Bleeding Edge not the Nightly build.

**Comment by JayFoxRox** [comment]
Wrong. PR2799 does not indicate version 2799. Also as @j-selby noted, this patch is not available in nightly builds yet. It is being tested in recent Bleeding Edge builds, but it might not be in all of them.

Can't you just wait a couple of weeks until the fix is ready for users and leave this issue-tracker to developers?
If you have questions about progress you should ask on the forums or discord (= user targeted), *not* github (= developer targeted).

**Comment by Gynlod23** [comment]
thanks bro

**Comment by IchigoMait** [comment]
@JayFoxRox Thx for clarifying it, I'm gonna not update Bleeding Edge then and stick to the version I have until this fix is out of Bleeding Edge.

**Comment by jroweboy** [comment]
@IchigoMait did you misunderstand what JayFoxRox said? The fix is in the most recent bleeding edge build and has been for the last 19 days. Please test it on bleeding edge and make sure that the problem is fixed. If it is fixed, then do *not* post about it, but if its is *not* fixed, then write a message here and paste your log from the crash as well. (In summary, test it out and report back here if it is not fixed for you)

**Comment by IchigoMait** [comment]
@jroweboy Are you illiterate? He didn't say what you said, learn to read, he said it's not in Nightly Builds even if the version number might be the same, he said it's being tested in the Bleeding Builds but it might not be in all versions. He said this comment section is meant for developers/contributors of the emulator.

Also there were no info about how to get this fix, I just assumed Bleeding Edge Build would probably have it, that's why let people know who come here like me and wonder how to get a "proper" fix.

Don't be such a douche.

**Comment by j-selby** [comment]
Bug has been fixed as of 045d0b5. Closing...
